### PR TITLE
[Ide] Allow dropping files from SolutionFolder to Project/ProjectFolder.

### PR DESCRIPTION
**Pull request by Therzok**
Bug 40873 - Cannot drag file from sln folder to project

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/FolderNodeBuilder.cs [hunk 1]
```diff
@@ -263,7 +265,19 @@ void DropNode (HashSet<SolutionItem> projectsToSave, object dataObject, HashSet<
 				projectsToSave.Add (targetProject);
 				return;
 			}
-			else
+			else if (dataObject is SolutionFolderFileNode) {
+				var sff = (SolutionFolderFileNode)dataObject;
+				if (sff != null) {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/FolderNodeBuilder.cs]: It can't be null

Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/FolderNodeBuilder.cs]: Good point.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/FolderNodeBuilder.cs [hunk 2]
```diff
@@ -263,7 +265,19 @@ void DropNode (HashSet<SolutionItem> projectsToSave, object dataObject, HashSet<
...
+					sff.Parent.Files.Remove (sff.FileName);
+
+					using (var monitor = IdeApp.Workbench.ProgressMonitors.GetStatusProgressMonitor ("Saving solution...", Stock.StatusSolutionOperation, true)) {
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/FolderNodeBuilder.cs]: Use ProjectOperations.SaveAsync(solution) for saving

**Comment by slluis** [comment]
Other than what I commented, it looks good.

**Comment by Therzok** [comment]
Manually merged. b9b75eed348afa29251f82a0d8791b4205eca24b

**Pull request closed without merging**